src/cli/deploy.ts

```diff
@@ -93,6 +93,35 @@ export async function getUserConfirmation(opts: { confirm: boolean }): Promise<b
   return confirm.confirm ? true : false;
 }
 
+async function buildAndDeployModule(image: string, force: boolean): Promise<void> {
+  const builtModule = await buildModule();
+  if (!builtModule) {
+    return;
+  }
+
+  // Generate a secret for the module
+  const webhook = new Assets(
+    { ...builtModule.cfg.pepr, description: builtModule.cfg.description },
+    builtModule.path,
+    [],
+  );
+  webhook.image = image ?? webhook.image;
+
+  try {
+    await webhook.deploy(deployWebhook, force, builtModule.cfg.pepr.webhookTimeout ?? 10);
+
+    // wait for capabilities to be loaded and test names
+    validateCapabilityNames(webhook.capabilities);
+
+    // Wait for the pepr-system resources to be fully up
+    await namespaceDeploymentsReady();
+    console.info(`✅ Module deployed successfully`);
+  } catch (e) {
+    console.error(`Error deploying module:`, e);
+    process.exit(1);
```

Comment:
Same question if we want to handle this differently than process.exit. Does not really have to do with updating peer deps though and it was already here. Could wait until that story comes up since we would have to handle the return on this function differently

Reply:
Agreed, we have #1380 for this specific goal so we'll leave it as-written on this PR.